tests/misc/tests/misc.js

```diff
@@ -640,4 +641,109 @@ describe("misc", () => {
+        new anchor.Wallet(anchor.web3.Keypair.generate()),
+        { commitment: program.provider.connection.commitment }
+      )
+    );
+    // Request airdrop for secondary wallet
+    const signature = await program.provider.connection.requestAirdrop(
+      anotherProgram.provider.wallet.publicKey,
+      anchor.web3.LAMPORTS_PER_SOL
+    );
+    await program.provider.connection.confirmTransaction(signature);
+    // Create all the accounts
+    await Promise.all([
+      program.rpc.testFetchAll(filterable1, {
+        accounts: {
+          data: data1.publicKey,
+          authority: program.provider.wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+        },
+        signers: [data1],
+      }),
+      program.rpc.testFetchAll(filterable1, {
+        accounts: {
+          data: data2.publicKey,
+          authority: program.provider.wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+        },
+        signers: [data2],
+      }),
+      program.rpc.testFetchAll(filterable2, {
+        accounts: {
+          data: data3.publicKey,
+          authority: program.provider.wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+        },
+        signers: [data3],
+      }),
+      anotherProgram.rpc.testFetchAll(filterable1, {
+        accounts: {
+          data: data4.publicKey,
+          authority: anotherProgram.provider.wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+        },
+        signers: [data4],
+      }),
+    ]);
+    // Call for multiple kinds of .all
+    const allAccounts = await program.account.dataWithFilter.all();
+    const allAccountsFilteredByBuffer =
+      await program.account.dataWithFilter.all(
+        program.provider.wallet.publicKey.toBuffer()
+      );
+    const allAccountsFilteredByBufferAndOffset =
+      await program.account.dataWithFilter.all(
+        program.provider.wallet.publicKey.toBuffer(),
+        1
+      );
+    const allAccountsFilteredByProgramFilters1 =
+      await program.account.dataWithFilter.all([
+        {
+          memcmp: {
+            offset: 8,
+            bytes: program.provider.wallet.publicKey.toBase58(),
+          },
+        },
+        { memcmp: { offset: 40, bytes: filterable1.toBase58() } },
+      ]);
+    const allAccountsFilteredByProgramFilters2 =
+      await program.account.dataWithFilter.all([
+        {
+          memcmp: {
+            offset: 8,
+            bytes: program.provider.wallet.publicKey.toBase58(),
+          },
+        },
+        { memcmp: { offset: 40, bytes: filterable2.toBase58() } },
+      ]);
+    // Without filters there should be 4 accounts
+    assert.equal(allAccounts.length, 4);
+    // Filtering by main wallet there should be 3 accounts
+    assert.equal(allAccountsFilteredByBuffer.length, 3);
+    // By changing the offset of the discriminator everything is filtered out
```

Comment:
Since the extra offset parameter is used for the discriminator filter it always returns empty. This is either a feature to allow future changes to the discriminator without needing to change the `all` method or it isn't working as expected.

Reply:
I see. I think you were correct to not include the offset parameter initially. It's not needed with the extra filters provided. Let's remove it?

Reply:
Consider it done.